### PR TITLE
Phone number generator

### PR DESCRIPTION
**Pull request by victor-dat21b**
Har fået lavet en masse tests til PhoneNumberGenerator.
De første test er til at teste at strukturen der bliver spyttet ud af PNG er korrekt. Disse er opdelt efter Equivalence Partitioning.

De efterfølgende er Boundary Value test med både 2 og 3 værdier. Jeg lavede 3 på dem der opstod som enkeltværdier i vores "validprefixes" og 2 Boundary approach til dem der fungere som intervaller. Testene er ikke grupperet efter Equivalence Partitioning, idet jeg følte det gav et bedre overblik at holde alle test for én værdi samlet, hvilket burde give bedre overlbik.

Jeg lagde mærke til at Bjørn har lavet hele klassen og dens metoder static static hvilket giver god mening, så jeg har indrettet mig efter dette og indkorporeret det.

### CODE REVIEW

File: UnitTests/PhoneNumberGeneratorTests.cs [hunk 1]
```diff
@@ -1,51 +1,509 @@
...
             // Act
-            string phoneNumber = PhoneNumberGenerator.GeneratePhoneNumber();
+            string PhoneNumber = PhoneNumberGenerator.GeneratePhoneNumber();
 
             // Assert
-            Assert.Matches(@"^\d+$", phoneNumber);
+            Assert.Matches(@"^\d+$", PhoneNumber);
+        }
+
+        //NEGATIVE General structure test 1/10
```
Comment (Troels-dat21B) [inline UnitTests/PhoneNumberGeneratorTests.cs]: Er ikke helt med på de tests der indeholde ikke "normale skrift tegn". Er forvirret over hvorfor de er med

**Review by Troels-dat21B — COMMENTED** [review]
Skal nok bare lige havde forklaret hvorfor de vestlige sprog med med i testene der tjekker skrift i nummeret

File: UnitTests/PhoneNumberGeneratorTests.cs [hunk 2]
```diff
@@ -1,51 +1,509 @@
...
+        //For the tests here, Black box test design technique has been used.
+        //First up there are tests that make sure the general structure should work. These has been ordered by Equivalence Partitioning.
+
+        //Then lastly the Boundary Value Analysis tests.
+        //There has been used 3 boundary values test cases for the single prefix entries, such as "441".
+        //We have run 3 boundary tests on 6 entries in the list and have a final of 18 tests.
+        //There has been used 2 boundary values test cases for the sequence of prexies, such as "344-349".
+        //We have run 2 boundary tests on 5 entries in the list and have a final of 20 tests.
+
+        string[] validPrefixes = PhoneNumberGenerator.validPrefixes;
```
Comment (Bjand49) [inline UnitTests/PhoneNumberGeneratorTests.cs]: I blackboxtesting må vi ikke tage listen af gyldige værdier fra klassen

File: UnitTests/PhoneNumberGeneratorTests.cs [hunk 3]
```diff
@@ -1,51 +1,509 @@
...
+            // Act & Assert
+            Assert.DoesNotContain("70", PhoneNumberGenerator.validPrefixes);
+        }
+
+        //NEGATIVE 3-Boundary test 3/18 - Invalid boundary above '71'
+        [Fact]
+        public void InvalidPhoneNumber_WithPrefixAbove71_ShouldFail()
+        {
+            // Act & Assert
+            Assert.DoesNotContain("72", PhoneNumberGenerator.validPrefixes);
```
Comment (Bjand49) [inline UnitTests/PhoneNumberGeneratorTests.cs]: Vær sikker på at du bruger den værdi der bliver sat i toppen, istedet for at bruge værdien fra PhoneNumberGenerator

**Review by Ceazer101 — DISMISSED** [review]

**Review by Bjand49 — CHANGES_REQUESTED** [review]
skal fikse kommentarer

File: UnitTests/PhoneNumberGeneratorTests.cs [hunk 4]
```diff
@@ -1,51 +1,183 @@
...
-        
+
+        //For the tests here, Black box test design technique has been used.
+        //First up there are tests that make sure the general structure should work. These has been ordered by Equivalence Partitioning.
+
+        //Then lastly the Boundary Value Analysis tests.
+        //There has been used 3 boundary values test cases for the single prefix entries, such as "441".
+        //We have run 3 boundary tests on 6 entries in the list and have a final of 18 tests.
+        //There has been used 2 boundary values test cases for the sequence of prexies, such as "344-349".
+        //We have run 2 boundary tests on 5 entries in the list and have a final of 20 tests.
```
Comment (Bjand49) [inline UnitTests/PhoneNumberGeneratorTests.cs]: du har stadig referencer til test amount

File: UnitTests/PhoneNumberGeneratorTests.cs [hunk 5]
```diff
@@ -1,51 +1,183 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Bjand49) [inline UnitTests/PhoneNumberGeneratorTests.cs]: it be empty down here eh

**Review by Bjand49 — CHANGES_REQUESTED** [review]

**Review by Bjand49 — DISMISSED** [review]

**Review by Bjand49 — APPROVED** [review]
